src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java

```diff
@@ -86,6 +86,20 @@
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings({ "PMD.LawOfDemeter" })
+    public String buildDirectoryPath(long dirId) throws MopsException {
+        String result = "";
+        Directory dir = getDirectory(dirId);
+        if (dir.getParentId() != null) {
+            result = buildDirectoryPath(dir.getParentId()) + "/" + dir.getName();
+        }
+        return result;
+    }
+
```

Comment:
can you make it so the root dir returns "/" and everything else returns something like "/foo/bar/"

Reply:
No - when you return something other than "" the whole material-path redirects to error 404. Debugging not successful, help greatly appreciated